### PR TITLE
feat(Navidrome): add presence

### PR DESCRIPTION
**Pull request by AlperShal**
## Description 
Added support for Navidrome, a self-hosted music streaming service.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
Nothing Playing:

image

Playing:
image

Paused:
image
</details>

### CODE REVIEW

File: websites/N/Navidrome/presence.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		else playerState = "paused";
+	} else playerState = "notInitialized";
+
+	// Check if music-player-panel initialized || Check if any music is playing/paused
+	if (playerState === "playing" || playerState === "paused") {
+		// Grab song information
+
+		const timestamps = presence.getTimestamps(
+			presence.timestampFromFormat(
+				document.querySelectorAll(".current-time")[0].innerHTML
```
Comment (DarkVillager) [inline websites/N/Navidrome/presence.ts:25]: If u get the first one from querySelectorAll then just use querySelector
```suggestion
				document.querySelectol(".current-time").innerHTML
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/Navidrome/presence.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		else playerState = "paused";
+	} else playerState = "notInitialized";
+
+	// Check if music-player-panel initialized || Check if any music is playing/paused
+	if (playerState === "playing" || playerState === "paused") {
+		// Grab song information
+
+		const timestamps = presence.getTimestamps(
+			presence.timestampFromFormat(
+				document.querySelectol(".current-time").innerHTML
```
Comment (DarkVillager) [inline websites/N/Navidrome/presence.ts:25]:
```suggestion
				document.querySelector(".current-time").innerHTML
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/Navidrome/metadata.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "263735224221827072",
+		"name": "alpershal"
+	},
+	"service": "Navidrome",
+	"description": {
+		"en": "Navidrome is a self-hosted, open source music server and streamer. It gives you freedom to listen to your music collection from any browser or mobile device."
+	},
+	"url": "navidrome.org",
```
Comment (theusaf) [inline websites/N/Navidrome/metadata.json:11]: Website supports www subdomain
```suggestion
	"url": [
		"navidrome.org",
		"www.navidrome.org"
	],
```

File: websites/N/Navidrome/presence.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/mhD2eIr.png",
+	};
+
+	// Determine player state
+	let playerState;
+	// Check if audio-title has any content
+	if (document.querySelectorAll(".audio-title")[0].innerHTML !== "") {
```
Comment (theusaf) [inline websites/N/Navidrome/presence.ts:13]: Prefer `.textContent` over `.innerHTML` where possible.
```suggestion
	if (document.querySelector(".audio-title").textContent) {
```

File: websites/N/Navidrome/presence.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		if (document.querySelectorAll(".react-jinke-music-player-pause-icon")[0])
```
Comment (theusaf) [inline websites/N/Navidrome/presence.ts:14]: `.querySelectorAll(...)[0]` can be simplified.
```suggestion
		if (document.querySelector(".react-jinke-music-player-pause-icon"))
```

File: websites/N/Navidrome/presence.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+			presence.timestampFromFormat(
+				document.querySelector(".current-time").innerHTML
+			),
+			presence.timestampFromFormat(
+				document.querySelector(".duration").innerHTML
+			)
+		);
+		// Set gathered data
+		presenceData.details = document.querySelectorAll(".songTitle")[0].innerHTML;
+		presenceData.state = document.querySelectorAll(".songArtist")[0].innerHTML;
```
Comment (theusaf) [inline websites/N/Navidrome/presence.ts:33]: Ditto

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**